include/boost/math/differentiation/autodiff.hpp

```diff
@@ -1291,12 +1288,11 @@ fvar<RealType,Order> sqrt(const fvar<RealType,Order>& cr)
   {
     root_type numerator = 0.5;
     root_type powers = 1;
-    derivatives[1] = numerator / *derivatives;
-    for (size_t i=2 ; i<=order ; ++i)
+    for (size_t i=1 ; i<=order ; ++i)
     {
-      numerator *= -0.5 * ((i<<1)-3);
-      powers *= x;
       derivatives[i] = numerator / (powers * *derivatives);
+      numerator *= root_type(-0.5 * (((i+1)<<1)-3));
+      powers *= x;
```

Comment:
Any comment on the performance implications of this? Seems like before optimizations you're performing these *=s redundantly once each where the previous solution didn't. Similarly you're comparing i to order one additional time with this approach. Does this all just get optimized away at the end of the day?

Reply:
There will be one extra multiplication/assignment each for `numerator`  and for `powers`. A conditional checking to see if we’re in the final iteration (so no such operations are needed) can be added. For arithmetic types, the current code would waste a small number of cycles, but it’s suboptimal to leave the code this way. 

Thanks for catching this!

Reply:
I updated the PR with a conditional around the two highlighted lines. Thanks again for catching this.